pkg/ccl/crosscluster/producer/replication_manager.go

```diff
@@ -137,20 +141,52 @@ func (r *replicationStreamManagerImpl) PlanLogicalReplication(
 
 	spans := make([]roachpb.Span, 0, len(req.TableIDs))
 	tableDescs := make([]descpb.TableDescriptor, 0, len(req.TableIDs))
+	typeDescriptors := make([]descpb.TypeDescriptor, 0)
+	foundTypeDescriptors := make(map[descpb.ID]struct{})
+	descriptors := r.txn.Descriptors()
 	for _, requestedTableID := range req.TableIDs {
-		td, err := r.txn.Descriptors().MutableByID(r.txn.KV()).Table(ctx, descpb.ID(requestedTableID))
+
+		td, err := descriptors.MutableByID(r.txn.KV()).Table(ctx, descpb.ID(requestedTableID))
 		if err != nil {
 			return nil, err
 		}
 		spans = append(spans, td.PrimaryIndexSpan(r.evalCtx.Codec))
 		tableDescs = append(tableDescs, td.TableDescriptor)
+
+		// fetch Type Descriptors, if any.
+		dbDesc, err := descriptors.MutableByID(r.txn.KV()).Database(ctx, td.GetParentID())
+		if err != nil {
+			return nil, err
+		}
+		typeIDs, _, err := td.GetAllReferencedTypeIDs(dbDesc,
+			func(id descpb.ID) (catalog.TypeDescriptor, error) {
+				if id > 0 {
```

Comment:
what's the deal with this function passed to `GetAllREferencedTypeIDs`? as the panic suggests, it doesn't seem to get called.